Observer/RestoreQuote.php

```diff
@@ -144,8 +149,8 @@ public function execute(\Magento\Framework\Event\Observer $observer)
                 if ($this->helper->getRestoreQuoteLastOrder() && ($lastRealOrder->getData('state') === 'new' && $lastRealOrder->getData('status') === 'pending') && $payment->getMethodInstance()->usesRedirect) {
 
                     //diactivate customer quotes
```

Comment:
what happens if I am a guest and already have items in cart and after that I click on the link in the second chance email?

Reply:
you will logout, and try login as customer which use second chance